### PR TITLE
feat: state management for creators

### PR DESCRIPTION
**Pull request by lauti7**
This PR: 
- adds endpoints for creators to handle Quest Instances state
- fixes db error namings
- fixes paginated endpoints to return the total items

### CODE REVIEW

File: crates/server/src/api/routes/quest_instances/get.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    match db.get_quest_instance(&quest_instance_id).await {
+        Ok(instance) => match db.is_quest_creator(&instance.quest_id, &address).await {
+            Ok(is_creator) if !is_creator => HttpResponse::from_error(QuestError::NotQuestCreator),
+            Ok(_) => HttpResponse::Ok().json(GetQuestInstanceResponse { instance }),
+            Err(err) => {
+                log::error!(
+                    "error on checking if {address} is quest creator of {}",
+                    instance.quest_id
+                );
```
Comment (aleortega) [inline crates/server/src/api/routes/quest_instances/get.rs:47]: Should we log this error? seems a business validation and may can cause flood on logs in we get attacked

Comment (aleortega) [reply crates/server/src/api/routes/quest_instances/get.rs:47]: Should we return Unauthorized on this case?

Comment (aleortega) [reply crates/server/src/api/routes/quest_instances/get.rs:47]: Or maybe return `HttpResponse::from_error(QuestError::NotQuestCreator)`

**Review by aleortega — APPROVED** [review]
LGTM

File: crates/server/src/api/routes/quest_instances/get.rs (hunk 1 above)
Comment (lauti7) [reply crates/server/src/api/routes/quest_instances/get.rs:47]: `HttpResponse::from_error(QuestError::NotQuestCreator)` is already considered in the first `Ok()` branch. The function returns an error when the DB fails, not when there is a mismatch.

**Pull request merged**